### DISCUSSION TITLE
Problem when compiling and running the code

### DISCUSSION THREAD

**Discussion by jamlemoos** [ERGO-Code/HiGHS#1242] [General]
Hello, I have a problem compiling and running the code. Does anyone have a step-by-step how to compile after cloning the project?

**Comment by jajhall** [comment]
What is your operating system, and do you want to call HiGHS from an application, or just run the executable?

**Reply by jamlemoos** [reply]
Hi Julian, thanks for the quick answer. I want to use HiGHS from a compiler. So, I am doing the following steps:

1 - I clone the project on the pc; 
2 - I go to the directory where I cloned; 
3 - I use mkdir build; 
4 - I use cd build; 
5 - I do cmake --build . 
6 - I try to compile using "g++ -o use_highs use_highs.cpp" in the build folder;

Then, when I do that it says that there is no executable file for use_highs.cpp or just highs.cpp in both the build and the HiGHS root. My PC uses the Ubuntu subsystem via WSL2.

**Comment by jajhall** [comment]
Thanks for this. Are you on Windows, MacOS or Linux?


--
Dr. J. A. Julian Hall, Reader, School of Mathematics,
University of Edinburgh, James Clerk Maxwell Building,
Peter Guthrie Tait Road, EDINBURGH, EH9 3FD, UK.
Room: 5418 Phone: [+44](131) 650 5075 Email: ***@***.***
Web: https://www.maths.ed.ac.uk/school-of-mathematics/people/a-z?person=47
My working hours may not be your working hours. Do not feel pressure to reply to this email outside your working hours.
My pronouns are he/him
________________________________

**Reply by jamlemoos** [reply]
I'm using Windows 10 and Ubuntu Subystem (linux) via WSL2.

**Reply by jajhall** [reply]
Thanks for the re-clarification of your OS: I was responding to the original reply in Portuguese

**Reply by jajhall** [reply]
I'm puzzled by your reference to `use_highs.cpp` or `highs.cpp`. There are no such files in the HiGHS repository. There is `src/lp_data/Highs.cpp`, but that isn't a `main` C++ file that would yield an executable.

The command `cmake --build` should yield the executable `build/bin/highs`, and that is how you run HiGHS from the command line

**Comment by jamlemoos** [comment]
Sorry, I ended up putting the wrong file here in the discussion. When I try to compile with both make and g++ it gives an error regarding the structure of the project, to be more precise, about how the files are being imported into each other. For example, when I try to compile the MIP solver I get this error: 

![image](https://user-images.githubusercontent.com/40551481/230469154-5d532401-7ef9-46aa-a228-f54af66cddd5.png)

I tried to change the import of some files but it only works when they are in the same folder, for example if it is in the MIP folder, I remove the "mip/" and leave only the file itself and then it works. But when there are files from other folders it also gives the error and I have tried to import in different ways but it always breaks. When I open from Visual Studio I can access everything correctly by clicking with cntrl on the import in the way it comes structured from git. I wonder if I have to change something about the structure in a makefile?

**Comment by jajhall** [comment]
HiGHS isn't used this way at all. 

When you run `cmake --build`, all the of the source code of HiGHS is built. After you've done this you can run HiGHS using the executable `bin/highs` or, if you want to call HiGHS from an application, link it using the library in `lib/`

There's a lot more help available in via https://ergo-code.github.io/HiGHS/dev/

If you've been changing the structure of the HiGHS source files, then I suggest that you delete what you've cloned or checked out, and start again.

**Comment by jamlemoos** [comment]
Hi, @jajhall, I did what you said and followed the step-by-step and was able to compile from the following form:

./bin/highs --solver=simplex ../check/instances/afiro.mps 

I saw that the solver options are only simplex and ipm, is there any way to run it using mip as the solver?

**Comment by jajhall** [comment]
HiGHS uses the MIP solver when the model it has is a MIP. 

`afiro` is an LP, when the choice between simplex and IPM is valid.

If you run

./bin/highs ../check/instances/flugpl.mps

then, because this is a MIP, you see the MIP solver output.
